### PR TITLE
Change default of float_precision for read_csv and read_table to "high"

### LINKED ISSUE #17154: read_csv returns different float values for same number
#### Code Sample, a copy-pastable example if possible
`test.csv`
```
-15.361
-15.361000
```

```
>>> import pandas as pd
>>> x = pd.read_csv('test.csv', header=None)
>>> x.loc[0, 0] == x.loc[1, 0]
False
```
#### Problem description / Expected output

The expected output of the code above is 
```
>>> x.loc[0, 0] == x.loc[1, 0]
True
```
We should expect both `-15.361` and `-15.361000` to be converted to the same `np.float64` representation. However, they are converted to different float values, differing in exactly the last bit of their floating point representation. For some reason, `-15.361` gets converted incorrectly to `0xC02EB8D4FDF3B645` whereas `-15.361000` is correctly to `0xC02EB8D4FDF3B646`.

For completeness, here are some more comparisons
`x.loc[1, 0]` is equal (`==`) to `np.float64('-15.361')`, `np.float64('-15.361000')`, and `float('-15.361000')`.
`x.loc[0, 0]` is not equal to any of those.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.3.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-73-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.3
pytest: None
pip: 9.0.1
setuptools: 36.2.4
Cython: None
numpy: 1.13.1
scipy: 0.19.1
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
pandas_gbq: None
pandas_datareader: None

### PR DESCRIPTION
**Pull request by Dr-Irv**
- [x] closes #17154
- [x] tests added / passed
   - modified `tests/io/parser/test_c_parser.py` to make sure all 4 options are tested
   - added `tests/io/parser/test_c_parser.py:test_high_is_default`
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry
  - for version 1.2

See discussion at bottom of #36149 for the performance tests.  Added `float_precision="legacy"` so people can pick up the old parser.  Can't change default to "high" because of incompatibility with python parser

### CODE REVIEW

**Comment by WillAyd** [comment]
If there is no performance difference should we not just get rid of the legacy parsing altogether?

**Comment by Dr-Irv** [comment]
> If there is no performance difference should we not just get rid of the legacy parsing altogether?

My only concern here is that maybe someone has code that inadvertently depends on it, so we have to keep it in there for some form of compatibility.  I do think we could deprecate the legacy parsing

**Comment by jreback** [comment]
yep that sounds fine to leave th option

just update the doc string to indicate

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -338,9 +338,9 @@
     option can improve performance because there is no longer any I/O overhead.
 float_precision : str, optional
     Specifies which converter the C engine should use for floating-point
-    values. The options are `None` for the ordinary converter,
-    `high` for the high-precision converter, and `round_trip` for the
-    round-trip converter.
+    values. The options are `None` or `high` for the ordinary converter,
+    `legacy` for the original lower precision pandas converter, and
+    `round_trip` for the round-trip converter.
```
Comment (Dr-Irv) [inline pandas/io/parsers.py:343]: @jreback doc string was updated already, so I think this is good to go??

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/io/parsers.py (hunk 1 above)
Comment (jreback) [reply pandas/io/parsers.py:343]: can you add a versionchanged tag for 1.2

File: pandas/tests/io/parser/test_c_parser_only.py [hunk 2]
```diff
@@ -646,7 +648,7 @@ def test_1000_sep_with_decimal(
     tm.assert_frame_equal(result, expected)
 
 
-@pytest.mark.parametrize("float_precision", [None, "high", "round_trip"])
+@pytest.mark.parametrize("float_precision", [None, "legacy", "high", "round_trip"])
```
Comment (jreback) [inline pandas/tests/io/parser/test_c_parser_only.py:651]: do we have a test that catches if in an invalid float_precision value is passed?

**Review by jreback — CHANGES_REQUESTED** [review]
minor comments

File: pandas/tests/io/parser/test_c_parser_only.py (hunk 2 above)
Comment (Dr-Irv) [reply pandas/tests/io/parser/test_c_parser_only.py:651]: - There is no such test
- Currently, in 1.1.1, if you give an incorrect value for `float_precision`, it treats it as `None`
```python
>>> df=pd.read_csv("../Misc/pandasDev/foo.csv", float_precision="yo")
>>> df
     A    B
0  NaN   B1
1  1.0  NaN
>>>

So I could add code to reject invalid values and then an appropriate test.  Let me know if I should do that.

Comment (jreback) [reply pandas/tests/io/parser/test_c_parser_only.py:651]: ok, can you add (and of course reject invalid values)

**Comment by Dr-Irv** [comment]
@jreback added the check for an invalid `float_precision` option and now all green

File: pandas/io/parsers.py [hunk 3]
```diff
@@ -2299,6 +2299,7 @@ def TextParser(*args, **kwds):
         values. The options are None for the ordinary converter,
         'high' for the high-precision converter, and 'round_trip' for the
         round-trip converter.
+        .. versionchanged:: 1.2
```
Comment (jreback) [inline pandas/io/parsers.py:2302]: check that tis renders ok, I *think* need a blank line after

**Review by jreback — APPROVED** [review]
great @Dr-Irv very minor comment can be addressed in a followon (if needed)

**Pull request merged**